domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1108 @@
+      }
+    }
+    //endregion
+
+    //region Perform the 'prompt' for a destination domain.
+    if ($policy_content === 'prompt' || $policy_users === 'prompt') {
+      // Make a list of the eligible destination domains in form id -> name.
+      $noassign_domain_ids = $delete_domain_ids + [$default_domain->id()];
+      $reassign_list = [
+        '0' => dt('Do not reassign'),
+        '1' => dt('Default domain'),
+      ];
+      $reassign_list = array_map(
+        function ($v) {
+          return $v->getHostname();
+        },
+        $this->filterDomains($all_domains, $noassign_domain_ids, $reassign_list)
+      );
+
+      if ($policy_content === 'prompt') {
+        $reassign_content = $this->io()
+                                 ->choice(dt('Reassign content to:'), $reassign_list);
+        if (empty($reassign_content)) {
+          throw new DomainCommandException('Cancelled.');
+        }
+        $policy_content = ($reassign_content == 0) ? 'ignore' : $reassign_content;
+      }
+      if ($policy_users === 'prompt') {
+        $reassign_users = $this->io()
+                              ->choice(dt('Reassign users to:'), $reassign_list);
+        if (empty($reassign_users)) {
+          throw new DomainCommandException('Cancelled.');
+        }
+        $policy_users = ($reassign_users == 0) ? 'ignore' : $reassign_users;
+      }
+    }
+    if ($policy_content === 'default') {
+      $policy_content = $default_domain->getDomainId();
+    }
+    if ($policy_users === 'default') {
+      $policy_users = $default_domain->getDomainId();
+    }
+    //endregion
+
+    //region Reassign content as required.
+    $options = [
+      'entity_filter' => 'node',
+      'policy' => $policy_content,
+      'multidomain' => FALSE,
+    ];
+    $this->reassignLinkedEntities($domains, $options);
+    $options = [
+      'entity_filter' => 'user',
+      'policy' => $policy_users,
+      'multidomain' => FALSE,
+    ];
+    $this->reassignLinkedEntities($domains, $options);
+    //endregion
+    
+    //$this->deleteDomain($domains);
+    //$this->logger()->info(dt('Domain record deleted.'));
+  }
+
+  /**
+   * Tests domains for proper response.
+   *
+   * If run from a subfolder, you must specify the --uri.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to test. If no value is passed,
+   *   all domains are tested.
+    * @param array $options An associative array of options whose values come
+    *   from cli, aliases, config, etc.
+   * @option base_path
+   *   The subdirectory name if Drupal is installed in a folder other than
+   *   server root.
+   * @usage drush domain-test
+   * @usage drush domain-test example.com
+   * @usage drush domain-test 1
+   *
+   * @command domain:test
+   * @aliases domain-test
+   * 
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function test($domain_id, array $options = ['base_path' => null]) {
```

Comment:
See `checkHTTPResponse` above, which should probably contain most of this checking code.